### PR TITLE
Deny duplicated row field names in MongoDB

### PR DESCRIPTION
**Pull request by Grandys**
## Description
Fixes https://github.com/trinodb/trino/issues/17812

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# MongoDB
* Validate if row has no duplicated fields when table is created ({issue}`17812`)
```

### CODE REVIEW

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java [hunk 1]
```diff
@@ -658,10 +660,37 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
...
+                .peek(MongoMetadata::validateRowFieldsForDuplicatedNamedFields)
                 .map(m -> new MongoColumnHandle(m.getName(), m.getType(), m.isHidden(), Optional.ofNullable(m.getComment())))
                 .collect(toList());
     }
 
+    private static void validateRowFieldsForDuplicatedNamedFields(ColumnMetadata column)
+    {
+        if (column.getType() instanceof RowType rowColumn) {
+            List<String> duplicatedFields = rowColumn.getFields()
+                    .stream()
```
Comment (ebyhr) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:672]: nit: Unwrap.

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java [hunk 2]
```diff
@@ -658,10 +660,37 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
     private static List<MongoColumnHandle> buildColumnHandles(ConnectorTableMetadata tableMetadata)
     {
         return tableMetadata.getColumns().stream()
+                .peek(MongoMetadata::validateRowFieldsForDuplicatedNamedFields)
                 .map(m -> new MongoColumnHandle(m.getName(), m.getType(), m.isHidden(), Optional.ofNullable(m.getComment())))
                 .collect(toList());
     }
 
+    private static void validateRowFieldsForDuplicatedNamedFields(ColumnMetadata column)
```
Comment (ebyhr) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:668]: Does this method work in nested row type? e.g. `a row(b row(c int, c int))`

File: plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java [hunk 3]
```diff
@@ -887,6 +887,17 @@ public void testSystemSchemas()
...
     }
 
+    @Test
+    public void testInvalidCreateTableWithDuplicatedRowFields()
+    {
+        assertQueryFails(
+                "create table t3 (a row(a int, a int, c int, b int))",
+                "Row field 'a' specified more than once");
+        assertQueryFails(
+                "create table t3 (a row(a int, a int, c int, b int, b int))",
```
Comment (ebyhr) [inline plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:897]: nit: Uppercase `create table`

File: plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java [hunk 4]
```diff
@@ -887,6 +887,17 @@ public void testSystemSchemas()
...
+                "Row fields 'a','b' specified more than once");
```
Comment (ebyhr) [inline plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:898]: Please create a table directly in MongoDB with `_schema` collection and try `INSERT` statement in Trino.

**Comment by ebyhr** [comment]
Could you update the commit title as `Deny duplicated row field names in MongoDB`? 
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages is our commit message guideline.

File: plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java (hunk 4 above)
Comment (Grandys) [reply plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:898]: 1. I added tests for case sensitivity and nested rows
2. For inserting table directly into schema, what's the expected outcome? incorrect results are returned (as they are now), or we validate insert or select if table definition is not correct?

Comment (Grandys) [reply plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:898]: I covered few more use cases, like row nested in array and map

Comment (ebyhr) [reply plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:898]: Regarding 2, insert/select should fail.

**Comment by mosabua** [comment]
@cla-bot check

**Comment by mosabua** [comment]
:wave: @Grandys @ebyhr  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Pull request closed without merging**